hack/scripts/patch-igz/patch_remote.py

```diff
@@ -0,0 +1,404 @@
+#!/usr/bin/env python3
+# Copyright 2024 The Nuclio Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import io
+import json
+import logging
+import os
+import shlex
+import subprocess
+import typing
+
+import click
+import coloredlogs
+import paramiko
+import yaml
+
+log_level = logging.INFO
+fmt = "%(asctime)s %(levelname)s %(message)s"
+logging.basicConfig(level=log_level)
+logger = logging.getLogger("nuclio-patch")
+coloredlogs.install(level=log_level, logger=logger, fmt=fmt)
+supported_targets = ["dashboard", "controller"]
+
+
+def target_wrapper(func):
```

Comment:
I’d give the decorator a more descriptive name to have over the methods, maybe `for_each_target` or `run_for_each_target` or `run_on_all_targets`